### PR TITLE
Homework_10 Hibernate DAO

### PR DESCRIPTION
**Pull request by IgorPrihodko**

### CODE REVIEW

File: src/main/java/dao/code/ConfirmationCodeDaoHibernate.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    @Override
+    public void addConfirmationCode(ConfirmationCode confirmationCode) {
+        Transaction transaction = null;
+        try(Session session = HibernateUtil.getSessionFactory().openSession()) {
+            transaction = session.beginTransaction();
+            session.saveOrUpdate(confirmationCode);
+            transaction.commit();
+        } catch (HibernateException e) {
+            logger.error("Can not add new confirmation code " + confirmationCode, e);
+            if (transaction != null) transaction.rollback();
```
Comment (SevostyanovGerman) [inline src/main/java/dao/code/ConfirmationCodeDaoHibernate.java:29]: {}

File: src/main/java/dao/code/ConfirmationCodeDaoHibernate.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        } catch (HibernateException e) {
+            logger.error("Can not remove confirmation code with id " + id, e);
+            if (transaction != null) transaction.rollback();
+        }
+    }
+
+    @Override
+    public List<ConfirmationCode> getByUserEmail(String userEmail) {
+        Transaction transaction = null;
+        try(Session session = HibernateUtil.getSessionFactory().openSession()) {
```
Comment (SevostyanovGerman) [inline src/main/java/dao/code/ConfirmationCodeDaoHibernate.java:56]: cs

File: src/main/java/dao/order/OrderDaoHibernate.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        } catch (HibernateException e) {
+            logger.error("Can not add new stockOnOrder " + stockOnOrder, e);
+            if (transaction != null) transaction.rollback();
+        }
+    }
+
+    @Override
+    public void removeOrder(Long id) {
+        Transaction transaction = null;
+        try(Session session = HibernateUtil.getSessionFactory().openSession()) {
```
Comment (SevostyanovGerman) [inline src/main/java/dao/order/OrderDaoHibernate.java:35]: cs

File: src/main/java/dao/order/OrderDaoImpl.java [hunk 4]
```diff
@@ -11,30 +11,30 @@
 public class OrderDaoImpl implements OrderDao {
 
     private static final Logger logger = Logger.getLogger(OrderDaoImpl.class);
-    private static final List<Order> orders = new ArrayList<>();
+    private static final List<StockOnOrder> STOCK_ON_ORDERS = new ArrayList<>();
```
Comment (SevostyanovGerman) [inline src/main/java/dao/order/OrderDaoImpl.java:14]: not const

File: src/main/java/model/Basket.java [hunk 5]
```diff
@@ -1,31 +1,65 @@
...
 public class Basket {
 
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id")
     private Long id;
-    private Long userID;
+
+    @JoinColumn(name = "user")
+    @ManyToOne(fetch = FetchType.LAZY, targetEntity = User.class, cascade = CascadeType.MERGE)
```
Comment (SevostyanovGerman) [inline src/main/java/model/Basket.java:30]: targetEntity why?

**Review by SevostyanovGerman — CHANGES_REQUESTED** [review]

File: src/main/java/dao/code/ConfirmationCodeDaoHibernate.java (hunk 2 above)
Comment (IgorPrihodko) [reply src/main/java/dao/code/ConfirmationCodeDaoHibernate.java:56]: fixed

File: src/main/java/dao/order/OrderDaoHibernate.java (hunk 3 above)
Comment (IgorPrihodko) [reply src/main/java/dao/order/OrderDaoHibernate.java:35]: fixed

File: src/main/java/dao/code/ConfirmationCodeDaoHibernate.java (hunk 1 above)
Comment (IgorPrihodko) [reply src/main/java/dao/code/ConfirmationCodeDaoHibernate.java:29]: fixed

File: src/main/java/dao/order/OrderDaoImpl.java (hunk 4 above)
Comment (IgorPrihodko) [reply src/main/java/dao/order/OrderDaoImpl.java:14]: fixed

File: src/main/java/model/Basket.java (hunk 5 above)
Comment (IgorPrihodko) [reply src/main/java/model/Basket.java:30]: fixed

**Pull request merged**